crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -16,7 +18,7 @@ pub struct TxId(pub H256);
 /// An ERC20 token address.
 ///
 /// https://eips.ethereum.org/EIPS/eip-20
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, From, Into)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, From, Into, Display)]
```

Comment:
`Display` automatically derives `ToString`. For `H160` the `ToString` does not print the full address, but it prints it in the format: `0x1234...ABCD` (the three dots are also printed like that, literally). For which purpose will we be using the `Display`? if we need to know the precise address, it's better to implement our own `Display` in which we guarantee the full address is displayed.

Reply:
Interesting. `Display` was added because token addresses were printed as part of the `InvalidPrice` errors.

If so, I removed the `Display` and printed the debug version which should be ok.